Simplenote/Classes/NoteOptionsViewController.swift

```diff
@@ -241,42 +292,177 @@ class NoteOptionsViewController: UITableViewController {
     }
 
-    func handleCopyLink() {
-        ///Handle copy link logic here
+    func handleCopyLink(from indexPath: IndexPath) {
+        guard !note.publishURL.isEmpty, let publishURL = URL(string: kSimplenotePublishURL + note.publishURL) else {
+            return
+        }
+
+        SPTracker.trackEditorPublishedUrlPressed()
+
+        let activityViewController = UIActivityViewController(activityItems: [publishURL],
+                                                              applicationActivities: [SPActivitySafari()])
+        if UIDevice.sp_isPad() {
+            activityViewController.modalPresentationStyle = .popover
+
+            let presentationController = activityViewController.popoverPresentationController
+            presentationController?.permittedArrowDirections = .any
+            presentationController?.sourceRect = tableView.rectForRow(at: indexPath)
+            presentationController?.sourceView = tableView
+        }
+        present(activityViewController,
+                animated: true,
+                completion: nil)
     }
 
-    func handleCollaborate() {
-        ///Handle collaboration logic here
+    func handleCollaborate(from indexPath: IndexPath) {
+        SPTracker.trackEditorCollaboratorsAccessed()
+
+        let collaboratorView = SPAddCollaboratorsViewController()
+        collaboratorView.collaboratorDelegate = self
+        collaboratorView.setup(withCollaborators: note.emailTagsArray as? [String])
+
+        let navController = SPNavigationController(rootViewController: collaboratorView)
+        navController.displaysBlurEffect = true
+
+        if UIDevice.sp_isPad() {
+            navController.modalPresentationStyle = .popover
+
+            let presentationController = navController.popoverPresentationController
+            presentationController?.permittedArrowDirections = .any
+            presentationController?.sourceRect = tableView.rectForRow(at: indexPath)
+            presentationController?.sourceView = tableView
+        }
+        present(navController,
+                animated: true,
+                completion: nil)
     }
 
     func handleMoveToTrash() {
-        ///Handle move to tash logic here
+        SPTracker.trackEditorNoteDeleted()
+        delegate?.didTapMoveToTrash(sender: self)
     }
 
     // MARK: - Navigation button handling
     @objc
     func handleDone(button: UIBarButtonItem) {
         dismiss(animated: true)
     }
+
+    // MARK: - Note saving
+    func save() {
+        note.modificationDate = Date()
+        note.createPreview()
+
+        SPAppDelegate.shared().save()
+        SPTracker.trackEditorNoteEdited()
+        CSSearchableIndex.default().indexSearchableNote(note)
+    }
+}
+
+// MARK: - Action protocol
+//
+/// A protocol to pass row actions to the parent view controller for handling
+protocol NoteOptionsViewControllerDelegate: class {
+    func didToggleMarkdown(toggle: UISwitch, sender: NoteOptionsViewController)
```

Comment:
Would the delegate need direct access to the internal UISwitch?

Reply:
I think I passed it just in case we needed to do anything to the switch but that's not the case in the end. I can just read the markdown state directly from `currentNote` and remove everything passed to the delegate